libdnf/module/ModulePackage.cpp

```diff
@@ -67,6 +78,54 @@ static void setSovable(Pool * pool, Solvable *solvable, std::string name,
     solvable_add_deparray(solvable, SOLVABLE_PROVIDES, depId, -1);
 }
 
+static std::string getPlatformIdFromQuery(Query & query)
+{
+    DnfSack * sack = query.getSack();
+    auto queryResult = query.runSet();
+    Id pkgId = -1;
+    std::string platform;
+    while ((pkgId = queryResult->next(pkgId)) != -1) {
+        std::unique_ptr<DnfPackage> pkg(dnf_package_new(sack, pkgId));
+        auto provides = dnf_package_get_provides(pkg.get());
+        auto count = provides->count();
+        for (int index = 0; index < count; ++index) {
+            Dependency provide(sack, provides->getId(index));
+            auto provideName = provide.getName();
+            size_t lenProvide = strlen(provideName);
+            if (lenProvide > 15 && strncmp("base-module(", provideName, 12) == 0
+                and provideName[lenProvide - 1] == ')') {
+                std::string tmpPlatform (provideName + 12, lenProvide - 13);
+                if (platform.empty()) {
+                    platform = tmpPlatform;
+                } else {
+                    if (platform != tmpPlatform) {
+                        auto logger(Log::getLogger());
+                        if (dnf_package_installed(pkg.get())) {
+                            logger->debug(
+                                _("Multiple module platforms provided by installed packages\n"));
+                        } else {
+                            logger->debug(
+                                _("Multiple module platforms provided by available packages\n"));
+                        }
+                        platform.clear();
+                        return platform;
+                    }
+                }
+            }
+        }
+    }
+    return platform;
+}
```

Comment:
Could we make this a more generic function? This would be useful in more than one place (releasever detection, control flags in packages, etc.).

For example, if I wanted to force switching from normal upgrade to dist-upgrade in PackageKit if I detected a certain flag during a upgrade transaction, similar to this: https://github.com/hughsie/PackageKit/commit/770ee092d13c0955401d3ef6414fb0f562079857